notebooks/Threshold Optimization Postprocessing for Binary Classification.ipynb

```diff
@@ -0,0 +1,458 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Getting and preparing the data\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To demonstrate the post processing algorithm we use the \"COMPAS\" dataset from [ProPublica](https://raw.githubusercontent.com/propublica/compas-analysis/master/compas-scores-two-years.csv). The labels represent the two-year recidivism ID, i.e. whether a person got rearrested within two years (label 1) or not (label 0). The features include sex, age, as well as information on prior incidents.\n",
+    "\n",
+    "To start, let's download the dataset using `tempeh`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "import numpy as np\n",
+    "from tempeh.configurations import datasets\n",
+    "\n",
+    "compas_dataset = datasets['compas']()\n",
+    "X_train = pd.DataFrame(compas_dataset.X_train, columns=compas_dataset.features)\n",
+    "y_train = pd.Series(compas_dataset.y_train.reshape(-1).astype(int), name=\"two_year_recid\")\n",
+    "X_test = pd.DataFrame(compas_dataset.X_test, columns=compas_dataset.features)\n",
+    "y_test = pd.Series(compas_dataset.y_test.reshape(-1).astype(int), name=\"two_year_recid\")\n",
+    "sensitive_features_train = pd.Series(compas_dataset.race_train)\n",
+    "sensitive_features_test = pd.Series(compas_dataset.race_test)\n",
+    "X_train.loc[0], y_train[0]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Create a fairness-unaware model"
```

Comment:
A little more context would be useful here

Reply:
I've changed things up a little bit and put more of the text descriptions before the code